variables.tf

```diff
@@ -1,36 +1,170 @@
-#   - Below are some common module input variables
-#   - They should be updated for input variables applicable to the module being added
-#   - Use variable validation when possible
-#
+variable "cluster_id" {
+  type        = string
+  description = "The ID of the cluster you wish to deploy the agent in"
+}
 
-variable "name" {
+variable "cluster_resource_group_id" {
   type        = string
-  description = "A descriptive name used to identify the resource instance."
+  description = "The Resource Group ID of the cluster"
 }
 
-variable "plan" {
+variable "cluster_config_endpoint_type" {
+  description = "Specify which type of endpoint to use for for cluster config access: 'default', 'private', 'vpe', 'link'. 'default' value will use the default endpoint of the cluster."
   type        = string
-  description = "The name of the plan type supported by service."
-  default     = "standard"
+  default     = "default"
+  nullable    = false # use default if null is passed in
   validation {
-    condition     = contains(["standard", "cos-one-rate-plan"], var.plan)
-    error_message = "The specified pricing plan is not available. The following plans are supported: 'standard', 'cos-one-rate-plan'"
+    error_message = "Invalid Endpoint Type! Valid values are 'default', 'private', 'vpe', or 'link'"
+    condition     = contains(["default", "private", "vpe", "link"], var.cluster_config_endpoint_type)
   }
 }
 
-variable "resource_group_id" {
+variable "is_vpc_cluster" {
+  description = "Specify true if the target cluster for the monitoring agent is a VPC cluster, false if it is a classic cluster."
+  type        = bool
+  default     = true
+}
+
+variable "wait_till" {
+  description = "To avoid long wait times when you run your Terraform code, you can specify the stage when you want Terraform to mark the cluster resource creation as completed. Depending on what stage you choose, the cluster creation might not be fully completed and continues to run in the background. However, your Terraform code can continue to run without waiting for the cluster to be fully created. Supported args are `MasterNodeReady`, `OneWorkerNodeReady`, `IngressReady` and `Normal`"
   type        = string
-  description = "The ID of the resource group where you want to create the service."
+  default     = "Normal"
+
+  validation {
+    error_message = "`wait_till` value must be one of `MasterNodeReady`, `OneWorkerNodeReady`, `IngressReady` or `Normal`."
+    condition = contains([
+      "MasterNodeReady",
+      "OneWorkerNodeReady",
+      "IngressReady",
+      "Normal"
+    ], var.wait_till)
+  }
+}
+
+variable "wait_till_timeout" {
+  description = "Timeout for wait_till in minutes."
+  type        = number
+  default     = 90
+}
+
+##############################################################################
+# Cloud Monitoring variables
+##############################################################################
+
+variable "cloud_monitoring_enabled" {
```

Comment:
there is no need to expose this boolean? This module only supports cloud monitoring agent, so a user would never set this to false